components/camel-graphql/src/main/java/org/apache/camel/component/graphql/GraphqlConfiguration.java

```diff
@@ -0,0 +1,82 @@
+package org.apache.camel.component.graphql;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+
+import org.apache.camel.RuntimeCamelException;
+import org.apache.camel.util.PropertiesHelper;
+import org.apache.camel.util.json.JsonObject;
+
+public class GraphqlConfiguration {
+
+    private static final String QUERY_PARAM = "query";
+    private static final String QUERY_FILE_PARAM = "queryFile";
+    private static final String OPERATION_NAME_PARAM = "operationName";
+    private static final String VARIABLE_PREFIX = "variable.";
+    private final URI httpUri;
+    private final String query;
+    private final String operationName;
+    private final String variables;
+
+    public GraphqlConfiguration(String remaining, Map<String, Object> parameters) {
+        this.httpUri = extractHttpUri(remaining);
+        this.query = extractQuery(parameters);
+        this.operationName = extractOperationName(parameters);
+        this.variables = extractVariables(parameters);
+    }
+
+    public URI getHttpUri() {
+        return httpUri;
+    }
+
+    public String getQuery() {
+        return query;
+    }
+
+    public String getOperationName() {
+        return operationName;
+    }
+
+    public String getVariables() {
+        return variables;
+    }
+
+    public String getRequestBody() {
+        JsonObject jsonObject = new JsonObject();
+        jsonObject.put("query", query);
+        jsonObject.put("operationName", operationName);
+        jsonObject.put("variables", variables);
+        return jsonObject.toJson();
+    }
+
+    private URI extractHttpUri(String remaining) {
+        try {
+            return new URI(remaining);
+        } catch (URISyntaxException e) {
+            throw new RuntimeCamelException("Invalid HTTP URI: " + remaining, e);
+        }
+    }
+
+    private String extractQuery(Map<String, Object> parameters) {
```

Comment:
If you let Camel do this, I don't think you will need it

Reply:
Done